src/styles/index.scss

```diff
@@ -1,3 +1,83 @@
+$day-size: 100px;
+$gap-size: 1px;
+$font-size: 30px;
+$day-bg: #eee;
+$hover-bg: #ffbfcb;
+$days: (
+  mon: 1,
+  tue: 2,
+  wed: 3,
+  thu: 4,
+  fri: 5,
+  sat: 6,
+  sun: 7,
+);
+
 body {
+  height: 100vh;
   margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  display: grid;
+  grid-template-columns: repeat(7, $day-size);
+  grid-auto-rows: $day-size;
+  gap: $gap-size;
+
+  @for $length from 28 through 30 {
+    &--month-length-#{$length} .calendar__day:nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
+}
+
+.calendar__day {
+  position: relative;
+  border: 1px solid black;
+  background-color: $day-bg;
+  box-sizing: border-box;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-family: Arial, sans-serif;
+  font-size: $font-size;
+  transition:
+    background-color 0.5s,
+    transform 0.5s;
+
+  &:hover {
+    cursor: pointer;
+    background-color: $hover-bg;
+    transform: translateY(-20px);
+  }
+
+  &::before {
```

Comment:
The `::before` pseudo-element is being used to insert the day numbers, but it has a `font-size` declaration that is unnecessary because you've already set the `font-size` for `.calendar__day`, which will be inherited by the `::before` pseudo-element.